src/Coordination/KeeperStorage.cpp

```diff
@@ -3104,7 +3040,8 @@ void KeeperStorage<Container>::preprocessRequest(
             return;
         }
 
-        new_deltas = preprocess(concrete_zk_request, *this, transaction->zxid, session_id, time, new_digest, *keeper_context);
+        uint64_t * digest_ptr = keeper_context->digestEnabled() ? &new_digest : nullptr;
```

Comment:
should we assert that `digest_ptr != nullptr` if `keeper_context->digestEnabled()`?

Reply:
in which place?
we manually set it to `nullptr` if it's not enabled here

Reply:
I meant after this line should we add `chassert(!keeper_context->digestEnabled() || new_digest)`